Android do have dprintf, getline and stpcpy in android-21,
and having custom definitions for them breaks on i686.

diff -u -r ../busybox-1.22.1/include/platform.h ./include/platform.h
--- ../busybox-1.22.1/include/platform.h	2014-01-19 21:38:10.000000000 -0500
+++ ./include/platform.h	2014-12-23 10:43:15.959769041 -0500
@@ -447,9 +447,6 @@
 #endif
 
 #if defined(ANDROID) || defined(__ANDROID__)
-# undef HAVE_DPRINTF
-# undef HAVE_GETLINE
-# undef HAVE_STPCPY
 # undef HAVE_STRCHRNUL
 # undef HAVE_STRVERSCMP
 # undef HAVE_UNLOCKED_LINE_OPS
